core/container_create.go

```diff
@@ -22,9 +22,10 @@ import (
 	"path/filepath"
 
 	"github.com/Mirantis/cri-dockerd/libdocker"
-	"github.com/docker/docker/api/types"
+	dockerbackend "github.com/docker/docker/api/types/backend"
```

Comment:
Hm... this feels "wrong". Did we not expose the right options in the API types? I _think_ the `backend`  is not expected to be used externally (should we consider making that `internal`?).

Reply:
Maybe it's expected in cri-dockerd though if it's implementing API and Backend (but I honestly would have to look at how cri-dockerd is implemented). Thought I'd at least leave a comment that this stood out to me.

Reply:
`dockerbackend.ContainerCreateConfig` is used here

https://github.com/Mirantis/cri-dockerd/blob/8ae7a0ee2a81ec928aee8a276ffe6ba0ec6125ba/libdocker/kube_docker_client.go#L143-L167

I guess it is possible to remove this dependency, but it is beyond the topic of this PR.

Reply:
Yes, it looks like the client (still) has this awkward signature to pass all those individual structs as separate arguments, but uses a local (non-exported) "ad-hoc" struct to create the equivalent for the request (grouping them in a single struct); https://github.com/moby/moby/blob/e61c425cc283ed85a6a87ab4750b52389aea4021/client/container_create.go#L15-L24

```go
type configWrapper struct {
	*container.Config
	HostConfig       *container.HostConfig
	NetworkingConfig *network.NetworkingConfig
}

// ContainerCreate creates a new container based on the given configuration.
// It can be associated with a name, but it's not mandatory.
func (cli *Client) ContainerCreate(ctx context.Context, config *container.Config, hostConfig *container.HostConfig, networkingConfig *network.NetworkingConfig, platform *ocispec.Platform, containerName string) (container.CreateResponse, error) {
	var response container.CreateResponse
```

https://github.com/moby/moby/blob/e61c425cc283ed85a6a87ab4750b52389aea4021/client/container_create.go#L75-L81

```go
	body := configWrapper{
		Config:           config,
		HostConfig:       hostConfig,
		NetworkingConfig: networkingConfig,
	}

	serverResp, err := cli.post(ctx, "/containers/create", query, body, **nil)
```


We should probably consider to either change that signature, and define a `CreateOptions` struct for the client, or (perhaps) for `cri-dockerd` to (for now) define a similar struct on its side for the time being

/cc @neersighted